src/Listener.cpp

```diff
@@ -143,9 +142,38 @@ std::string Listener::ThreadID()  {
 bool Listener::NotificationReceived() {
    std::vector<std::string> dataFromQueue;
    mQueueReader->GetShot(getShotTimeout, dataFromQueue);
-   return MessageHasPayload(dataFromQueue);
+   bool notificationReceived = MessageHasPayload(dataFromQueue);
+   ClearMessages();
```

Comment:
Why?  if notification was not received,. i.e. we had a timeout should it be cleared then?